File: src/services/webapi/src/modules/posts/dto/PostUpdate.dto.ts
import { Type } from 'class-transformer';
import { IsString, MinLength, ValidateNested } from 'class-validator';
import { AddressUpdateDto } from './AddressUpdate.dto';

export class PostUpdateDto {
  @IsString()
  @MinLength(3)
  title?: string;
  @IsString()
  @MinLength(5)
  content?: string;
  @ValidateNested()
  @Type(() => AddressUpdateDto)
  address?: AddressUpdateDto;
}

File: src/clients/android/app/src/main/java/com/example/student_community/models/webapi/dto/PostUpdateDto.kt
package com.example.student_community.models.webapi.dto

import com.google.gson.annotations.SerializedName

data class PostUpdateDto(
    @SerializedName("title") val Title: String?=null,
    @SerializedName("content") val Content: String?=null,
    @SerializedName("address") val Address: AddressUpdateDto?=null,
)